src/http.cpp

```diff
@@ -916,13 +922,25 @@ Request::handle_response_header(const asio::error_code& ec)
         }
     }
     // server wants to keep sending or we have content-length defined
-    else if (response_.headers[HTTP_HEADER_CONNECTION] == HTTP_HEADER_CONNECTION_KEEP_ALIVE)
+    else if (connection_it != response_.headers.end() and
+             response_.headers[HTTP_HEADER_CONNECTION] == HTTP_HEADER_CONNECTION_KEEP_ALIVE)
     {
-        conn_->async_read_until(JSON_VALUE_DELIM,
+        conn_->async_read_until(BODY_VALUE_DELIM,
             std::bind(&Request::handle_response_body, this, std::placeholders::_1, std::placeholders::_2));
     }
     else if (connection_type_ == restinio::http_connection_header_t::close)
+    {
         terminate(asio::error::eof);
+    }
+    else if (transfer_encoding_it != response_.headers.end() and
+             response_.headers[HTTP_HEADER_TRANSFER_ENCODING] == HTTP_HEADER_TRANSFER_ENCODING_CHUNKED)
```

Comment:
use iterator to access map value:
*transfer_encoding_it == HTTP_HEADER_TRANSFER_ENCODING_CHUNKED